File: examples/C/Data_Management_range_dynamic_vector_addition_exercise.c
#include <stdio.h>
#include <stdlib.h>
int main(void){
    int size = 10000;
    int a[size], b[size], c[size];

// Structured data region
    #pragma acc data create(a, b) copyout(c)
    {         
        #pragma acc parallel loop present(a, b)
        for(int i=0; i<size; ++i){
            a[i] = i;
            b[i] = 2*i;
        }

        #pragma acc parallel loop present(a, b, c)
        for (int i=0; i<size; ++i){
            c[i] = a[i]+b[i];
        }
    }
    printf("value at position 12: %d\n", c[12]);
}       

File: examples/Fortran/Data_Management_range_dynamic_vector_addition_exercise.f90
program range_dynamic
    use iso_fortran_env, only : INT32, REAL64
    implicit none
    integer(kind=INT32), parameter              :: system_size = 10000
    integer(kind=INT32), dimension(system_size) :: a, b, c
    integer(kind=INT32)                         :: i

    !$acc data create(a(:), b(:)) copyout(c(:))
    !$acc parallel loop present(a(:), b(:))
    do i = 1, system_size
        a(i) = i
        b(i) = i*2
    enddo

    !$acc parallel loop present(a(:), b(:), c(:))
    do i = 1, system_size
        c(i) = a(i) + b(i)
    enddo
    !$acc end data

    write(0,"(a21,i5)") "Value at position 12: ", c(12)
end program
